### PR TITLE
Add W&B lr logging and project naming

### PR DESCRIPTION
**Pull request by morganmcg1**
Adds:
-  logging of learning rate to Weights & Biases in `device_train.py`
-  ability to change the W&B project name in the config if needed (for `device_train` and `train.py`)
- `wandb_project` to the configs
- `name` and `wandb_project` description to doc string
- updates wandb version to >=11.2

Screenshot 2021-08-08 at 13 08 17

### CODE REVIEW

File: device_train.py [hunk 1]
```diff
@@ -312,4 +315,8 @@ def eval_step(network, data):
             steps_per_sec = 1 / (time.time() - start)
             tokens_per_sec = tokens_per_step * steps_per_sec
 
-            wandb.log({'train/loss': loss, 'train/last_loss': last_loss, 'train/steps_per_sec': steps_per_sec, 'train/tokens_per_sec': tokens_per_sec}, step)
+            wandb.log({'train/loss': loss, 
+                'train/last_loss': last_loss, 
+                'train/steps_per_sec': steps_per_sec, 
+                'train/tokens_per_sec': tokens_per_sec,
+                'train/learning_rate': float(scheduler(step))}, step)
```
Comment (nostalgebraist) [inline device_train.py:322]: I have a slight preference for using `network.state["opt_state"][-1].count().item()` in place of `step` here.

Why: the former is the actual step value optax uses when computing the lr.  It gets saved into checkpoints and loaded from them.

It _should_ always match the value of this script's `step` unless there's a bug.  But if there _is_ a bug, I'd prefer to report the real (and probably unintended) lr rather than the user-intended lr.

**Comment by kingoflolz** [comment]
@morganmcg1 Thanks for making this PR, agree with @nostalgebraist about the use of optax step count. Happy to merge once that's updated and merge conflict is resolved

**Comment by morganmcg1** [comment]
Thanks @kingoflolz, @nostalgebraist for the comments, conflict addressed. Also, wandb_project was moved from the arguments to the config since your last review.

The PR now adds:
-  logging of learning rate to Weights & Biases in `device_train.py`
-  ability to change the W&B project name in the config (for `device_train` and `train.py`)
- `wandb_project` to the existing config files
- `name` and `wandb_project` description to doc string
- updates `wandb` version to >=11.2

File: device_train.py (hunk 1 above)
Comment (morganmcg1) [reply device_train.py]: thanks, yep better to be safe than sorry! fixed

**Comment by kingoflolz** [comment]
Thanks!
